documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,226 @@ parameters through - see <<lucene-indexing-performance>> for more details.
+
+Hibernate Search provides a new mass indexing approach using batch job introduced by JSR352.
+It covers not only the existing functionality of the previous indexer, but also benefits the
+powerful standard based facilities of the Java Batch Platform (JSR-352), such as checkpoint and
+step partitioning. However, it requires a batch runtime for the execution:
+
+* For Java EE users, we use the batch runtime provided by your Java EE container.
+* For Java SE users, we use the batch runtime
+  https://github.com/jberet/jsr352/tree/master/jberet-se[JBeret SE].
+
+.How to use the new mass-indexer?
+====
+[source, JAVA]
+----
+Properties jobProps = MassIndexingJob
+        .parameters()
+        .forEntity( MyEntity.class )
+        .build();
+long executionId = BatchRuntime
+        .getJobOperator()
+        .start( MassIndexingJob.NAME, jobProps );
+----
+====
+
+In the following paragraphs, we will explain why JSR-352 standard is a good choice, and how to
+configure different parameters in order to bootstrap the mass-indexing job.
+
+==== Define your indexing scope
+Our job allows you to define your own indexing scope—you can start a full indexation or a partial
+indexation through 3 different methods: selecting the desired entity types, using HQL, or using
+Hibernate criteria.
+
+.How to add define the indexing scope in different methods?
+====
+[source, JAVA]
+----
+// full indexation
+Properties jobProps1 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps2 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( "select c from MyClass c where c.name in ( 'Foo', 'Bar' )" )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps3 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( Restrictions.in( "name", "Foo", "Bar" ) )
+        .build();
+----
+====
+
+==== Parallelism
+In order to maximize the performance, we highly recommend you to speed up the mass indexer using
+parallelism. Parallelism is activated by default. Under the standard JSR 352, the exact word is
+_"partitioning"_: The indexing step may run as multiple partitions, one per thread. Each partition
+has its own partition ID and parameters. If there’re more partitions than threads, partitions are\
+considered as a queue to consume: each thread can only run one partition at a time and won’t consume
+the next partition until the previous one is finished.
+
+===== Threads
+The maximum number of threads used by the job execution is defined through method `maxThreads()`.
+Within the N threads given, there’s 1 thread reserved for the core, so only N - 1 threads are
+available for different partitions. The default number of threads used in Hibernate Search is 10,
+which correspond to the max number of threads in JBeret. You can overwrite you maximum number used
```

Comment:
Please don't refer to JBeret in the docs, it's just one implementation among many others. Referring to it makes users of other JSR-352 implementations think that we just target JBeret, which isn't true.